test/org/zaproxy/zap/extension/websocket/client/ServerConnectionEstablisherUnitTest.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2018 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.zaproxy.zap.extension.websocket.client;
+
+import fi.iki.elonen.NanoWSD;
+import org.apache.commons.httpclient.URI;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.extension.websocket.ExtensionWebSocket;
+import org.zaproxy.zap.extension.websocket.WebSocketProxy;
+import org.zaproxy.zap.extension.websocket.utility.WebSocketUtils;
+import org.zaproxy.zap.testutils.WebSocketTestUtils;
+import org.zaproxy.zap.testutils.websocket.server.NanoWebSocketConnection;
+import org.zaproxy.zap.testutils.websocket.server.NanoWebSocketTestServer;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import java.util.Stack;
+
+public class ServerConnectionEstablisherUnitTest extends WebSocketTestUtils {
+	private final static int WEBSOCKET_PORT = 8885;
```

Comment:
We should use a random port (0) instead, otherwise this will cause problems if already in use or when running the tests in parallel.